src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11956,6 +11996,85 @@ private ExpressionSyntax ParseNewExpression()
+
+            var list = _pool.AllocateSeparated<CollectionElementSyntax>();
+
+            if (this.CurrentToken.Kind != SyntaxKind.CloseBracketToken)
+            {
+tryAgain:
+                if (this.IsPossibleExpression() || this.CurrentToken.Kind == SyntaxKind.CommaToken)
+                {
+                    list.Add(this.ParseCollectionElement());
+
+                    int lastTokenPosition = -1;
+                    while (IsMakingProgress(ref lastTokenPosition))
+                    {
+                        if (this.CurrentToken.Kind == SyntaxKind.CloseBracketToken)
+                        {
+                            break;
+                        }
+                        else if (this.IsPossibleExpression() || this.CurrentToken.Kind == SyntaxKind.CommaToken)
+                        {
+                            list.AddSeparator(this.EatToken(SyntaxKind.CommaToken));
+
+                            // check for exit case after legal trailing comma
+                            if (this.CurrentToken.Kind == SyntaxKind.CloseBraceToken)
+                            {
+                                break;
+                            }
+                            else if (!this.IsPossibleExpression())
+                            {
+                                goto tryAgain;
+                            }
+
+                            list.Add(this.ParseCollectionElement());
+                            continue;
+                        }
+                        else if (SkipBadCollectionCreationExpressionTokens(ref openBracket, list, SyntaxKind.CommaToken) == PostSkipAction.Abort)
+                        {
+                            break;
+                        }
+                    }
+                }
+                else if (SkipBadCollectionCreationExpressionTokens(ref openBracket, list, SyntaxKind.CommaToken) == PostSkipAction.Continue)
+                {
+                    goto tryAgain;
+                }
+            }
+
+            return _syntaxFactory.CollectionCreationExpression(
+                openBracket,
+                _pool.ToListAndFree(list),
+                this.EatToken(SyntaxKind.CloseBracketToken));
+        }
+
+        private PostSkipAction SkipBadCollectionCreationExpressionTokens(ref SyntaxToken openBracket, SeparatedSyntaxListBuilder<CollectionElementSyntax> list, SyntaxKind expected)
+        {
+            return this.SkipBadSeparatedListTokensWithExpectedKind(ref openBracket, list,
+                p => p.CurrentToken.Kind != SyntaxKind.CommaToken && !p.IsPossibleExpression(),
+                p => p.CurrentToken.Kind == SyntaxKind.CloseBracketToken || p.IsTerminator(),
+                expected);
+        }
+
+        private CollectionElementSyntax ParseCollectionElement()
+        {
+            var dotDotToken = this.TryEatToken(SyntaxKind.DotDotToken);
+
+            var expression = this.ParseExpression();
```

Comment:
This line causes a debug check to fail inside `ParseExpression`:

![image](https://user-images.githubusercontent.com/8040367/212560031-9178649e-7e56-4ff1-a15c-76a4d4be44d3.png)

Reply:
Went with `ParseExpressionCore` from context.

Reply:
yup.  that seems right :)